File: examples/v2/deployment-settings/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
project_name = config.get("projectName") or "my-new-project"
stack_name = config.get("stackName") or "dev"
executor_image = config.get("executorImage") or "pulumi-cli"

parent_stack = ps_v2.Stack(
    "parentStack",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
)

settings = ps_v2.DeploymentSettings(
    "settings",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
    executor_context={"executorImage": executor_image},
    operation_context={
        "preRunCommands": ["yarn"],
        "environmentVariables": {"TEST_VAR": "foo"},
        "options": {"skipInstallDependencies": True},
    },
    source_context={
        "git": {
            "repoUrl": "https://github.com/example/example.git",
            "branch": "refs/heads/main",
        },
    },
    opts=pulumi.ResourceOptions(depends_on=[parent_stack]),
)

pulumi.export("stackId", settings.stack_name)

File: examples/v2/deployment-settings/go/main.go
package main

import (
	v2 "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "my-new-project"
		}
		stackName := cfg.Get("stackName")
		if stackName == "" {
			stackName = "dev"
		}
		executorImage := cfg.Get("executorImage")
		if executorImage == "" {
			executorImage = "pulumi-cli"
		}

		parentStack, err := v2.NewStack(ctx, "parentStack", &v2.StackArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
		})
		if err != nil {
			return err
		}

		settings, err := v2.NewDeploymentSettings(ctx, "settings", &v2.DeploymentSettingsArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
			ExecutorContext: pulumi.Map{
				"executorImage": pulumi.String(executorImage),
			},
			OperationContext: pulumi.Map{
				"preRunCommands":       pulumi.StringArray{pulumi.String("yarn")},
				"environmentVariables": pulumi.StringMap{"TEST_VAR": pulumi.String("foo")},
				"options": pulumi.Map{
					"skipInstallDependencies": pulumi.Bool(true),
				},
			},
			SourceContext: pulumi.Map{
				"git": pulumi.Map{
					"repoUrl": pulumi.String("https://github.com/example/example.git"),
					"branch":  pulumi.String("refs/heads/main"),
				},
			},
		}, pulumi.DependsOn([]pulumi.Resource{parentStack}))
		if err != nil {
			return err
		}

		ctx.Export("stackId", settings.StackName)
		return nil
	})
}
